### PR TITLE
DOC: Update documentation on optimize.rosen

### PR DESCRIPTION
**Pull request by juliusbierk**
Documentation currently implies that function only takes 1-D arrays.
This PR adds a clarifying line.

### CODE REVIEW

**Comment by andyfaff** [comment]
It's not clear to me why the extra documentation should be added. I'm guessing that this behaviour might be useful for vectorisation? This behaviour also works for `rosen_der`, but it doesn't work for `rosen_hess`.
I don't necessarily think that the function was designed with this functionality in mind, so I'm not sure extra documentation is warranted here, especially because `rosen_hess` doesn't work in the same way.

**Comment by juliusbierk** [comment]
I can see how this behaviour not working for `rosen_hess` is a problem.

However, in particular for teaching optimization techniques it is useful for people to plot functions (I made this PR only because students of mine were confused by the signature being different from other mathematical function`scipy` functions). 
For visualisation it is very useful to be able to do

```
x = np.linspace(-3, 3, 50)
X, Y = np.meshgrid(x, x)
plt.plot_surface(X, Y, rosen([X, Y]))
```

but I see how this is not the main purpose of this function.

**Comment by ev-br** [comment]
From a teaching perspective, IMO:
- a single sentence added in this PR us too cryptic to be helpful for students
- an example from the comment above _is_ useful. 
- one more example with a 1D plot of one cross-section would be helpful to demonstrate a "typical" usage.

So overall I suggest reworking this PR into adding two examples with plots. The second example should also explicitly warn about `rosen_hess`.

**Comment by juliusbierk** [comment]
@ev-br I would be very for such a solution, however I suppose `scipy` documentation does not include plots or anything that depends on other libraries (other than `numpy`).

**Comment by ev-br** [comment]
You definitely can include plots in the Examples section, see e.g. https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.least_squares.html
Click on the Source link to see the numpydoc structure.

**Comment by juliusbierk** [comment]
Ah nice!

Well, ok! Should I do that then? :)

**Comment by juliusbierk** [comment]
Done.

File: scipy/optimize/optimize.py [hunk 1]
```diff
@@ -276,6 +276,7 @@ def rosen(x):
     ----------
     x : array_like
         1-D array of points at which the Rosenbrock function is to be computed.
+        n-dimensional arrays broadcast using the first component as variables.
```
Comment (andyfaff) [inline scipy/optimize/optimize.py:279]: I don't understand what "first component as variables" means.

Comment (juliusbierk) [reply scipy/optimize/optimize.py:279]: How about:
```n-dimensional arrays broadcast using the first dimension as function variables.```
?

**Comment by juliusbierk** [comment]
Also, circleci is giving me

```Traceback (most recent call last):
  File "/home/circleci/repo/venv/lib/python3.7/site-packages/matplotlib/sphinxext/plot_directive.py", line 472, in run_code
    exec(code, ns)
  File "<string>", line 2
    Note that ``rosen_hess`` does not broadcast in this manner.
            ^
SyntaxError: invalid syntax
```

Any idea of the problem?

File: scipy/optimize/optimize.py (hunk 1 above)
Comment (ev-br) [reply scipy/optimize/optimize.py:279]: ISTM this can be dropped altogether, the example that follows is plenty enough.

Comment (juliusbierk) [reply scipy/optimize/optimize.py:279]: ok!

**Comment by juliusbierk** [comment]
@andyfaff, @ev-br 
Ok with this now?

**Comment by ev-br** [comment]
LGTM now

**Pull request merged**